tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/PublicAssetMethodContract.scala

```diff
@@ -0,0 +1,659 @@
+package com.linagora.tmail.james.common
+
+import com.linagora.tmail.james.common.PublicAssetMethodContract.UploadResponse
+import io.netty.handler.codec.http.HttpHeaderNames.ACCEPT
+import io.restassured.RestAssured.{`given`, requestSpecification}
+import io.restassured.http.ContentType.JSON
+import net.javacrumbs.jsonunit.JsonMatchers
+import net.javacrumbs.jsonunit.assertj.JsonAssertions.assertThatJson
+import org.apache.http.HttpStatus
+import org.apache.james.GuiceJamesServer
+import org.apache.james.jmap.api.model.IdentityId
+import org.apache.james.jmap.http.UserCredential
+import org.apache.james.jmap.rfc8621.contract.Fixture.{ACCEPT_RFC8621_VERSION_HEADER, ACCOUNT_ID, ANDRE, ANDRE_PASSWORD, BOB, BOB_PASSWORD, DOMAIN, authScheme, baseRequestSpecBuilder}
+import org.apache.james.utils.DataProbeImpl
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.{BeforeEach, Test}
+import play.api.libs.json.{JsString, Json}
+
+import scala.jdk.CollectionConverters._
+
+object PublicAssetMethodContract {
+  case class UploadResponse(blobId: String, contentType: String, size: Long)
+}
+
+trait PublicAssetMethodContract {
+  @BeforeEach
+  def setUp(server: GuiceJamesServer): Unit = {
+    server.getProbe(classOf[DataProbeImpl])
+      .fluent()
+      .addDomain(DOMAIN.asString)
+      .addUser(BOB.asString(), BOB_PASSWORD)
+      .addUser(ANDRE.asString(), ANDRE_PASSWORD)
+
+    requestSpecification = baseRequestSpecBuilder(server)
+      .setAuth(authScheme(UserCredential(BOB, BOB_PASSWORD)))
+      .addHeader(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .build()
+      .log().ifValidationFails()
+  }
+
+  private def uploadAsset(content: Array[Byte] = "Your asset content here".getBytes,
+                          contentType: String = "image/png"): UploadResponse = {
+    val uploadResponse: String = `given`()
+      .basePath("")
+      .contentType(contentType)
+      .body(content)
+    .when
+      .post(s"/upload/$ACCOUNT_ID")
+    .`then`
+      .extract
+      .body
+      .asString
+
+    val blobId: String = (Json.parse(uploadResponse) \ "blobId").as[String]
+    val size: Long = (Json.parse(uploadResponse) \ "size").as[Long]
+    val assetType: String = org.apache.james.mailbox.model.ContentType.of((Json.parse(uploadResponse) \ "type").as[String])
+      .mimeType().asString()
+    UploadResponse(blobId, assetType, size)
```

Comment:
Idiomatic:

```
UploadResponse(blobId = (Json.parse(uploadResponse) \ "blobId").as[String], 
  assetType = (Json.parse(uploadResponse) \ "size").as[Long], 
  size = org.apache.james.mailbox.model.ContentType.of((Json.parse(uploadResponse) \ "type").as[String])
      .mimeType().asString())
```

BTW why doing `String => MimeType => String` ? Why not mandating the stronger type?

IE

```
case class UploadResponse(blobId: String, contentType: MimeType, size: Long)
```

&&

```
UploadResponse(blobId = (Json.parse(uploadResponse) \ "blobId").as[String], 
  assetType = (Json.parse(uploadResponse) \ "size").as[Long], 
  size = org.apache.james.mailbox.model.ContentType.of((Json.parse(uploadResponse) \ "type").as[String])
      .mimeType())
```

Reply:
> BTW why doing String => MimeType => String ? Why not mandating the stronger type?

UploadResponse is just a very simple POJO that helps with contract tests (that was choice replace to Tuple3)

When using Scala refined types, some assertJ methods don't support it, then we still need one step convert it to plain type